Xlib/protocol/rq.py

```diff
@@ -604,7 +604,8 @@ def parse_value(self, val, display):
         return self.type.parse_value(val, display)
 
     def pack_value(self, val):
-        return self.type.pack_value(val)
+        val = self.type.pack_value(val)
+        return val, len(val), None
```

Comment:
This comment in PR #224 is not addressed. What is the reason for this change? I thought we should fix `pack_value` implementation for specific type `self.type`, not for general `Object`. Or please explain if I'm thinking wrong way.

Reply:
See #223 for more details of my suggestions.

Reply:
Maybe it's worth separating pull requests for Randr 1.5 and `pack_value`.

Reply:
> Maybe it's worth separating pull requests for Randr 1.5 and `pack_value`.

Agreed. I'll pull this out.

Reply:
It looks like this change never was merged in a separate PR.

Without this change `set_monitor` is unusable, Ex:
```python
from Xlib import display
from Xlib.ext import randr

root = display.Display().screen().root
m = randr.get_monitors(root, is_active=True).monitors[0]
m.x = 500
randr.set_monitor(root, monitor_info=m)
```
raises "ValueError: too many values to unpack (expected 3)"

Also, more generally....
```python
from Xlib.protocol import rq
MonInfo = rq.Struct(
    rq.Int16('x'),
    rq.Int16('y'),
    rq.List('foo', rq.Int32),
)
TestStruct = rq.Struct(
    rq.Int16("a"),
    rq.Object('monitor_info', MonInfo),
)
TestStruct.to_binary(a=0, monitor_info=dict(x=1, y=0, foo=[1, 2, 3]))
```
fails in the same way.

The problem only occurs when the inner `Struct` contains a list.  If you remove the `List` from `MonInfo` then there's no exception.